projects/packages/stats/src/class-wpcom-stats.php

```diff
@@ -399,6 +404,44 @@ protected function fetch_stats( $args = array() ) {
 		return $wpcom_stats;
 	}
 
+	/**
+	 * Fetches stats data from WPCOM or local Cache. Caches locally for 5 minutes.
+	 *
+	 * Unlike the above function, this caches data in the post meta table. As such,
+	 * it prevents wp_options from blowing up when retrieving views for large numbers
+	 * of posts at the same time. However, the final response is the same as above.
+	 *
+	 * @param array $args Query parameters.
+	 * @param int   $post_id Post ID to acquire stats for.
+	 *
+	 * @return array|WP_Error
+	 */
+	protected function fetch_post_stats( $args, $post_id ) {
+		$endpoint    = $this->build_endpoint();
+		$meta_name   = '_' . self::STATS_CACHE_TRANSIENT_PREFIX;
+		$stats_cache = get_post_meta( $post_id, $meta_name );
+
+		if ( $stats_cache ) {
+			$data = reset( $stats_cache );
+
+			if ( is_wp_error( $data ) ) {
```

Comment:
> I took a look at the returning value of get_post_meta, which wouldn't be an array of WP_Error, so I guess the condition here will never be satisfied.

Ah, totally missed this. I've changed it now so that we only cache the stats data that we're interested in.

Reply:
ah sorry! I would prefer the errors being cached please just in case. I was thinking whether WP_Error could be serialized and unserialized correctly, but I just tested, it worked fine!

Could we revert the change please?

Reply:
No problem - switched it back!